addons/mail/static/src/js/attachment_box.js

```diff
@@ -65,6 +67,66 @@ var AttachmentBox = Widget.extend({
         this.currentResModel = record.model;
     },
 
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * Calls controller to upload file, separate method is created
+     * to use it in tests to patch it.
+     *
+     * @param {string} controllerUrl
+     * @param {FormData} formData
+     * @private
+     */
+    async _callUploadAttachment(controllerUrl, formData) {
+        const response = await window.fetch(controllerUrl, {
+            method: 'POST',
+            body: formData,
+        });
+        const result = await response.text();
+        const $el = $(result);
+        $.globalEval($el.contents().text());
+    },
+    /**
+     * Making sure that dragging content is external files.
+     * Ignoring other content draging like text.
+     *
+     * @private
+     * @param {DataTransfer} dataTransfer
+     * @returns {boolean}
+     */
+    _isDragSourceExternalFile: function (dataTransfer) {
```

Comment:
shorthand method name (`_isDragSourceExternalFile(dataTransfer) {}`)
(to do a bit everywhere in the diff)